storage/innobase/xtrabackup/src/keyring_plugins.cc

```diff
@@ -46,6 +46,30 @@ const char *XTRABACKUP_KEYS_FILE = "xtrabackup_keys";
 const char *XTRABACKUP_KEYS_MAGIC = "KEYSV01";
 const size_t XTRABACKUP_KEYS_MAGIC_SIZE = 7;
 
+/** Load plugins and keep argc and argv untouched */
+static void
+init_plugins(int argc, char **argv)
+{
+	int t_argc = argc;
+	char **t_argv = new char *[t_argc + 1];
+
+	memset(t_argv, 0, sizeof(char *) * (t_argc + 1));
+	memcpy(t_argv, argv, sizeof(char *) * t_argc);
+
+	mysql_optional_plugins[0] = 0;
+	mysql_mandatory_plugins[0] = 0;
+
+	if (opt_xtra_plugin_dir != NULL) {
+		strcpy(opt_plugin_dir, opt_xtra_plugin_dir);
```

Comment:
While this code is old, just moved, I don't see why we know that the buffer `opt_plugin_dir` points to is enough for these strings - shouldn't this be `strdup` / freeing the current buffer?

Reply:
it is a buffer of FN_REFLEN size, will change to strncpy